diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -89,42 +95,91 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     private lateinit var positionByOffset: (Int) -> Pair<Int, Int>
 
     override fun logic(node: ASTNode) {
-        if (node.elementType == KtFileElementType.INSTANCE) {
-            node.getChildren(null).forEach {
-                if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
-                    checkLength(it, configuration)
+        var currentFixNumber = 0
+        var isFixedSmthInPreviousStep: Boolean
+
+        // loop that trying to fix LineLength rule warnings until warnings run out
+        do {
+            isFixedSmthInPreviousStep = false
+            currentFixNumber++
+
+            if (node.elementType == KtFileElementType.INSTANCE) {
+                node.getChildren(null).forEach {
+                    if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
+                        val isFixedSmthInChildNode = checkLength(it, configuration)
+
+                        if (!isFixedSmthInPreviousStep && isFixedSmthInChildNode) {
+                            isFixedSmthInPreviousStep = true
+                        }
+                    }
                 }
             }
-        }
+        } while (isFixedSmthInPreviousStep && currentFixNumber < MAX_FIX_NUMBER)
```

Comment:
I guess we need to log or warn via diktat some error what we reached `MAX_FIX_NUMBER`